.pipelines/e2e.yml

```diff
@@ -10,47 +10,79 @@ resources:
 # Azure DevOps Pipeline running e2e tests
 variables:
 - template: vars.yml
-jobs:
-- job: E2E
-  timeoutInMinutes: 180
-  pool:
-    name: ARO-CI
-    demands: go-1.16
-  steps:
-  - template: ./templates/template-checkout.yml
-  - template: ./templates/template-az-cli-login.yml
-    parameters:
-      azureDevOpsJSONSPN: $(aro-v4-e2e-devops-spn)
-  - script: |
-      az account set -s $AZURE_SUBSCRIPTION_ID
-      export SECRET_SA_ACCOUNT_NAME=$(SECRET_SA_ACCOUNT_NAME)
-      make secrets
-      . secrets/env
-      echo "##vso[task.setvariable variable=RP_MODE]$RP_MODE"
-    displayName: 🔑 Downloading certificates and secrets from storage account
-    name: setEnv
-  - template: ./templates/template-push-images-to-acr.yml
-    parameters:
-      rpImageACR: $(RP_IMAGE_ACR)
-      buildCommand: publish-image-aro
-  - script: |
-      set -e
-      set -o pipefail
-
-      . secrets/env
-
-      az account set -s $AZURE_SUBSCRIPTION_ID
-
-      set -x
-      . ./hack/e2e/run-rp-and-e2e.sh
-      trap 'set +e; kill_rp; clean_e2e_db' EXIT
-
-      deploy_e2e_db
-
-      run_rp
-      validate_rp_running
-      register_sub
-
-      export CI=true
-      make test-e2e
-  - template: ./templates/template-az-cli-logout.yml
+
+# Run the test suite and collect must-gather
+stages:
+- stage: E2ESuite
+  jobs:
+  - job: Execute_Tests
+    timeoutInMinutes: 180
+    pool:
+      name: ARO-CI
+      demands: go-1.16
+    steps:
+    - template: ./templates/template-checkout.yml
+    - template: ./templates/template-az-cli-login.yml
+      parameters:
+        azureDevOpsJSONSPN: $(aro-v4-e2e-devops-spn)
+    - template: ./templates/template-certificates-and-secrets.yml
+    - template: ./templates/template-push-images-to-acr.yml
+      parameters:
+        rpImageACR: $(RP_IMAGE_ACR)
+        buildCommand: publish-image-aro
+    - script: |
+        set -o pipefail
+
+        . secrets/env
+        az account set -s $AZURE_SUBSCRIPTION_ID
+
+        set -x
+        . ./hack/e2e/run-rp-and-e2e.sh
+        trap 'set +e; kill_rp; clean_e2e_db' EXIT
+
+        deploy_e2e_db
+        run_rp
+        validate_rp_running
+        register_sub
+
+        export CI=true
+        export LOCATION=eastus
+        export BUILD_BUILDID=$(Build.BuildId)
+        export CLUSTER="v4-e2e-V$BUILD_BUILDID-$LOCATION"
+        export DATABASE_NAME="v4-e2e-V$BUILD_BUILDID-$LOCATION"
+        . secrets/env
+        . ./hack/e2e/run-rp-and-e2e.sh
+
+        if make test-e2e;
+        then
+          delete_e2e_cluster
+          exit 0
+        else
+          wget https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.9.8/openshift-client-linux-4.9.8.tar.gz
+          tar xf openshift-client-linux-4.9.8.tar.gz
+
+          make admin.kubeconfig
```

Comment:
The `make admin.kubeconfig` and `delete_e2e_cluster` command from the helper file both fail with the below. In e2e, the RP db is being created in our dev subscription and I've confirmed that the cluster document exists inside of it, but even when steps are in the same pipeline stage, they get a 404:
```
+ make admin.kubeconfig
hack/get-admin-kubeconfig.sh /subscriptions/46626fc5-476d-41ad-8c76-2ec49c6994eb/resourceGroups/e2einfra-eastus/providers/Microsoft.RedHatOpenShift/openShiftClusters/v4-e2e-V53835780-eastus >admin.kubeconfig
time="2022-04-04T16:47:19Z" level=info msg="running in local development mode" func="env.NewCore()" file="pkg/env/core.go:37"
time="2022-04-04T16:47:19Z" level=info msg="creating development InstanceMetadata" func="instancemetadata.New()" file="pkg/util/instancemetadata/instancemetadata.go:60"
time="2022-04-04T16:47:19Z" level=info msg="InstanceMetadata: running on AzurePublicCloud" func="env.NewCore()" file="pkg/env/core.go:45"
time="2022-04-04T16:47:22Z" level=fatal msg="404 : " func="main.main()" file="hack/db/db.go:83"
```
We can see in the above request (`hack/get-admin-kubeconfig.sh /subscriptions/46626fc5-476d-41ad-8c76-2ec49c6994eb/resourceGroups/e2einfra-eastus/providers/Microsoft.RedHatOpenShift/openShiftClusters/v4-e2e-V53835780-eastus >admin.kubeconfig`) that the cluster name and resource group are correct. The requests for kubeconfig and cluster deletion *should* be going to the same local RP that was used to build the cluster and run e2e tests.

The full build can be found here:

https://msazure.visualstudio.com/5d69ab04-7ded-49dc-84d5-bbbcac4add8d/_apis/build/builds/53835780/logs/13